src/app/static/src/tests/projects/actions.test.ts

```diff
@@ -651,19 +651,20 @@ describe("Projects actions", () => {
             currentVersion: mockProject.versions[0]
         });
 
-        const stateUrl = "/project/1/rename";
-        mockAxios.onPost(stateUrl, "name=renamedProject")
+        const stateUrl = "/project/1/rename?name=renamedProject&note=test%20notes";
```

Comment:
This line and the rest of the test would benefit from some tidying:
1. If you `await` on the action then you don't need the `setTimeout()`
2. `stateUrl` works but isn't correct - the `name` and `note` are actually sent as a form-encoded POST body, not via the URL
3. You should check that nothing is committed. This is important because the action calls `getCurrentProject` and `getProjects` regardless of whether any errors occur. Perhaps this should be resolved in the action itself, but it should definitely be checked here.

Feel to pick and choose from this version:
```typescript
    it("renameProject changes a project's name", async () => {
        const commit = jest.fn();
        const dispatch = jest.fn();
        const state = mockProjectsState({
            currentProject: mockProject,
            currentVersion: mockProject.versions[0]
        });

        mockAxios.onPost("/project/1/rename")
            .reply(200, mockSuccess("OK"));

        const projectPayload = {
            projectId: 1,
            name: "renamedProject",
            note: "test notes"
        }
        await actions.renameProject({commit, state, rootState, dispatch} as any, projectPayload);

        const posted = mockAxios.history.post[0].data;
        expect(posted).toEqual("name=renamedProject&note=test%20notes");
        expect(commit.mock.calls.length).toBe(0);
        expect(dispatch.mock.calls[0][0]).toBe("getCurrentProject");
        expect(dispatch.mock.calls[1][0]).toBe("getProjects");
        expect(dispatch.mock.calls.length).toBe(2);
    });
```